### DISCUSSION TITLE
Expose Studio components

### DISCUSSION THREAD

**Discussion by donaldpipowitch** [chakra-ui/panda#3014] [Ideas]
### Description

Hi, we'd like to be able to leverage @pandacss/studio in other contexts (e.g. Storybook). As far as I can tell it should be "relatively" simple by exporting the base components (e.g. [`packages/studio/src/components/colors.tsx`](https://github.com/chakra-ui/panda/blob/main/packages/studio/src/components/colors.tsx)) so that they can be used in bigger React applications.

The only thing (?) that I would assume that needs to be done is adding some kind of context provider to expose the tokens (- I guess this would be about rewriting [`packages/studio/src/lib/panda-context.ts`](https://github.com/chakra-ui/panda/blob/main/packages/studio/src/lib/panda-context.ts) to be an actual React Context that you provide the tokens to...?).

### Problem Statement/Justification

It makes `@pandacss/studio` more versatile and avoids re-creating the Studio functionality in community land.

### Proposed Solution or API

- Make components like [`packages/studio/src/components/colors.tsx`] a top-level export of `@pandacss/studio`.
- Provide a React Context Provider to be able to pass all the data that these components need at runtime.

### Alternatives

- Re-create the Studio functionality in user land.
- Live with the Astro based Studio next to other design/documentation artifacts.

**Comment by cgatian** [comment]
I thought the same thing. I loaded up Studio and thought it was a great resource, but storybook is going to be my source of truth. I really don't want to have two separate URLs for Storybook and Studio. Exposing the components would be a great start.

**Comment by segunadebayo** [comment]
Totally agree. We should consider exposing utilities for making your own studio within Storybook or design system docs.

**Reply by cgatian** [reply]
@segunadebayo I'm looking into building out Storybook "Doc Blocks" for Panda consumers. What's the best way to gain access to the generated tokens? I see there's a TokenDictionary package, is that going the right way?

**Comment by ambethia** [comment]
I have a font loaded from `next/font/google` in my app. I see the font documented in Studio, but the font itself is not loaded, presumably because there's nowhere I've configured it for Panda to find it (outside of my app layout).

Is there any existing way to configure Panda Studio to pickup my font?

I'm tacking this onto the conversation here about Storybook, because I presume if we can expose these elements in Storybook, they may work as intended?

**Reply by cgatian** [reply]
This is probably a separate issue unrelated to exposing the components.

**Reply by ambethia** [reply]
I agree this is probably a separate issue, but if anyone else stumbles here like I did, trying to solve this, I figured it out. There's a config option in the panda config that let's you inject a string into the head or body of the studio page:

```
  studio: {
    inject: {
      head: `<link href="https://fonts.googleapis.com/css2?family=Recursive:slnt,wght,CASL,CRSV,MONO@-15..0,300..800,0..1,0..1,0..1&display=swap" rel="stylesheet">
             <style>:root { --font-recursive: Recursive, sans-serif }</style>`,
    },
  },
```

Would still love to see these design tokens all together in Storybook.
